src/hotspot/share/opto/vectornode.cpp

```diff
@@ -1275,7 +1276,8 @@ Node* ReductionNode::make_reduction_input(PhaseGVN& gvn, int opc, BasicType bt)
 bool ReductionNode::implemented(int opc, uint vlen, BasicType bt) {
   if (is_java_primitive(bt) &&
       (vlen > 1) && is_power_of_2(vlen) &&
-      Matcher::vector_size_supported(bt, vlen)) {
+      Matcher::vector_size_supported(bt, vlen) &&
+      (vlen * type2aelembytes(bt) <= SuperWordMaxVectorSize)) {
```

Comment:
Can you put this whole condition into separate `static bool VectorNode::vector_size_supported(vlen, bt)` and use in both cases?